src/winkerberos.c

```diff
@@ -419,6 +433,163 @@ sspi_client_init(PyObject* self, PyObject* args, PyObject* kw) {
     return resultobj;
 }
 
+static PyObject*
+sspi_server_init(PyObject* self, PyObject* args, PyObject* kw) {
+	sspi_server_state* state;
+	PyObject* pyctx = NULL;
+	PyObject* serviceobj;
+	PyObject* principalobj = Py_None;
+	LONG flags = ISC_REQ_MUTUAL_AUTH | ISC_REQ_SEQUENCE_DETECT;
+	PyObject* userobj = Py_None;
+	PyObject* domainobj = Py_None;
+	PyObject* passwordobj = Py_None;
+	PyObject* mechoidobj = Py_None;
+	WCHAR *service = NULL, *principal = NULL;
+	WCHAR *user = NULL, *domain = NULL, *password = NULL;
+	Py_ssize_t slen, len, ulen, dlen, plen = 0;
+	WCHAR *mechoid = GSS_MECH_OID_KRB5;
+	PyObject* resultobj = NULL;
+	INT result = 0;
+	static SEC_CHAR* keywords[] = {
+		"service", "principal", "gssflags", "user", "domain", "password", "mech_oid", NULL };
```

Comment:
This project mimics the API of https://github.com/apple/ccs-pykerberos. It looks like the only parameter you want here is "service". I'm not sure what principal or gssflags is used for on the server side. On the bright side that means a huge amount of code can just go away.

Note that, though PyKerberos supports "DELEGATE" as a special service name, I'm not sure Winkerberos can actually support it because it doesn't seem like SSPI supports it.

https://github.com/apple/ccs-pykerberos/blob/PyKerberos-1.3.0/src/kerberos.c#L522

Reply:
It's actually just a straight copy of what's found in `sspi_client_init`. So you're right a lot of the code can go away.